src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/MultilineDetector.java

```diff
@@ -116,7 +116,7 @@ private void findMatch() {
             // OK http://blog.igorminar.com/2008/05/catching-stackoverflowerror-and-bug-in.html
             // http://programmers.stackexchange.com/questions/
             //        209099/is-it-ever-okay-to-catch-stackoverflowerror-in-java
-            options.getReporter().log(1, MSG_STACKOVERFLOW, matcher.pattern().toString());
+            options.getReporter().log(1, MSG_STACKOVERFLOW);
         }
```

Comment:
These changes are actually wrong https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/src/main/resources/com/puppycrawl/tools/checkstyle/checks/regexp/messages.properties#L8
because we need to mention the pattern here is required

But I think we can cover this only while we have some real input. I have tried some way on my local but it doesn't work

Reply:
I am not sure whether this type of case is possible in real code

Reply:
It was added at this commit https://github.com/checkstyle/checkstyle/commit/6f089534ea0b9680f0c76bc1fb3c04d11954a278

Please recheck PR and issue

Reply:
Do we still have test that was created originally? Recheck git history if it was removed or updated.

Can we extend it to validate full message? It is ok cover this case by pure Junit test. It has no much value for users, it is extreme edge case.

Reply:
https://github.com/checkstyle/checkstyle/blob/8134680099d82c13c8231ca1b96048a01773a356/src/test/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpMultilineCheckTest.java#L182-L196
Yes the test still exist.

Reply:
Can we extend message verification?

Reply:
![image](https://github.com/checkstyle/checkstyle/assets/97679350/13a6b1b3-b3ae-40f6-ba8f-0c3f98d46db6)
Here, their is something fault in code itself I am trying to find the cause. why this is happening may be becuase of exception

Reply:
If you fail to extend, push in GitHub what you tried, to let mentors to help easier.

Reply:
I have not make any changes just change a minor change in exisiting test

I have update the branch

Reply:
Please fix https://github.com/checkstyle/checkstyle/blob/d832cae442e754f6d76729ff4794cb6c6302a971/src/main/resources/com/puppycrawl/tools/checkstyle/checks/regexp/messages.properties#L8
To use index of parameter 0 instead of 1.

```
[ERROR]   RegexpMultilineCheckTest.testNoStackOverflowError:196->AbstractModuleTestSupport.verify:312->AbstractModuleTestSupport.verify:350->AbstractModuleTestSupport.verify:372->AbstractModuleTestSupport.verify:402->AbstractModuleTestSupport.lambda$verify$1:405 Violations for /tmp/junit13558589523051257823/junit13264957617699487808.tmp differ.
value of: iterable.onlyElement()
expected: 1: java.util.regex.Matcher caused a java.lang.StackOverflowError for pattern (x|y)* . (you may be scanning a binary file instead of text?)
but was : 1: java.util.regex.Matcher caused a java.lang.StackOverflowError for pattern '{1}' (you may be scanning a binary file instead of text?).
```

Reply:
done